socketio/src/client/reconnect.rs

```diff
@@ -248,4 +248,56 @@ mod test {
         socket.disconnect()?;
         Ok(())
     }
+
+    #[test]
+    fn socket_io_iterator_integration() -> Result<()> {
```

Comment:
@nshaaban-cPacket Is this test for iterator OK? ReconnectClient have to be manual created to not call `poll_callback` function.